src/Microsoft.Sbom.Api/Workflows/SbomConsolidationWorkflow.cs

```diff
@@ -92,15 +95,38 @@ private IEnumerable<ConsolidationSource> GetSbomsToConsolidate(string artifactPa
             return null;
         }
 
-        return detectedSboms.Select((sbom) => new ConsolidationSource(info, sbomConfigFactory.Get(sbom.manifestInfo, manifestDirPath, metadataBuilderFactory)));
+        return detectedSboms.Select((sbom) => new ConsolidationSource(info, sbomConfigFactory.Get(sbom.manifestInfo, manifestDirPath, metadataBuilderFactory), artifactPath));
     }
 
     private async Task<bool> ValidateSourceSbomsAsync(IEnumerable<ConsolidationSource> consolidationSources)
     {
-        var validationWorkflows = consolidationSources
-            .Select(async sbom => await Task.FromResult(true)); // TODO: Run validation workflow
-        var results = await Task.WhenAll(validationWorkflows);
-        return results.All(b => b);
+        var result = true;
+        foreach (var source in consolidationSources)
+        {
+            var identifier = Math.Abs(string.GetHashCode(source.SbomConfig.ManifestJsonFilePath)).ToString();
+            var defaultOutputPath = configuration.OutputPath.Value;
+            var workflowResult = false;
+            try
+            {
+                configuration.ValidateSignature.Value = !source.ArtifactInfo.SkipSigningCheck ?? true;
+                configuration.IgnoreMissing.Value = source.ArtifactInfo.IgnoreMissingFiles ?? false;
+                configuration.BuildDropPath.Value = source.BuildDropPath;
+                configuration.OutputPath.Value = fileSystemUtils.GetTempFile($"validation-results-{identifier}.json");
```

Comment:
For debugging, would be useful to create a temp folder, then put the files into that folder? That seems like it might be simpler to look through a few files than to try to filter from many. It also might make it a bit easier if we wanted to clean up the files after execution completes

Reply:
The generation code will need to add a wrapper of [Directory.CreateTempSubDirectory](https://learn.microsoft.com/en-us/dotnet/api/system.io.directory.createtempsubdirectory?view=net-8.0). Maybe there's some synergy here, maybe not